### PR TITLE
New mounted files source, content, and noExpand properties

### PR DESCRIPTION
**Pull request by ghen**
Added new properties support to mounted files spec:
* `source`
* `content`
* `noExpand`

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New chore (expected functionality to be implemented)

#### Checklist:

- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [x] I've signed off with an email address that matches the commit author.

### CODE REVIEW

File: internal/humanitec/convert.go [hunk 1]
```diff
@@ -84,8 +86,70 @@ func getProbeDetails(probe *score.ContainerProbeSpec) map[string]interface{} {
...
+			sb.WriteString(fmt.Sprint(str))
+		}
+		return sb.String(), nil
+		// END (TODO)
+	default:
+		return "", fmt.Errorf("can not use '%T' as a content for '%s': not supported", content, target)
+	}
+}
+
+// convertFileMountSpec extracts a mount file detailf from the source spec.
```
Comment (chrishumanitec) [inline internal/humanitec/convert.go:125]: nit: typo `detailf`

File: internal/humanitec/convert.go [hunk 2]
```diff
@@ -84,8 +86,70 @@ func getProbeDetails(probe *score.ContainerProbeSpec) map[string]interface{} {
...
+	if f.Source != "" {
+		content, err = readFile(baseDir, f.Source)
+	} else {
+		content, err = mergeFileContent(f.Content, f.Target)
+	}
+	if err != nil {
+		return "", nil, err
+	}
+
+	if !f.NoExpand {
```
Comment (chrishumanitec) [inline internal/humanitec/convert.go:139]: We actually need to do more here. Placeholders themselves need to be escaped if `NoExpand` is true.

E.g. with `NoExpand = true` the following should happen:

`${do-not-interpret}` -> `$\{do-not-intrepret}` As described here: https://developer.humanitec.com/platform-orchestrator/reference/placeholders

**Review by chrishumanitec — CHANGES_REQUESTED** [review]
If `NoExpand` is true, we need to escape placeholders, otherwise they will be expanded by Humanitec.

The usecase here is: including something like a raw bash script with `${SOME_VALUE}` that we want to be passed literally into the output file.

File: internal/humanitec/convert.go (hunk 2 above)
Comment (ghen) [reply internal/humanitec/convert.go:139]: I would disagree. The `noExpand` switch controls `score-humanitec` CLI tool behavior. It should not modify the source file.

If the User wants to control how Humanitec passes through the values, they should escape those in Score file explicitly. E.g. if the User would want to achieve this by editing values in Humanitec web UI, what value would they put into the edit box? Same value they should use in Score file.

File: internal/humanitec/convert.go (hunk 1 above)
Comment (ghen) [reply internal/humanitec/convert.go]: fixed

File: internal/humanitec/convert.go (hunk 2 above)
Comment (chrishumanitec) [reply internal/humanitec/convert.go:139]: That is not the way to think about it. The idea here is that the user sepciofies in the score file that *no escaping happens* In general they do not know what score implementation will be used. From the view of the user score-humanitec = Humanitec so the behaviour in teh score file is the outcome they are looking for not the input to Humanitec.

**Review by chrishumanitec — APPROVED** [review]
LGTM

**Pull request merged**